test/evaluation.spec.js

```diff
@@ -255,6 +255,13 @@ module.exports.addTests = function({testRunner, expect}) {
       const a = await page.evaluate(() => Array(100 * 1024 * 1024 + 1).join('a'));
       expect(a.length).toBe(100 * 1024 * 1024);
     });
+    it('should throw error with detailed information on exception inside promise ', async({page, server}) => {
+      let error = 'did not catch error';
+      await page.evaluate(() => new Promise(() => {
+        does_not_exist.click();
```

Comment:
i think this test would be easier to read if you just throw an error here

Reply:
Done.